packages/slice-machine/src/legacy/components/Forms/RenameVariationModal/RenameVariationModal.tsx

```diff
@@ -34,9 +34,47 @@ export const RenameVariationModal: FC<RenameVariationModalProps> = ({
   variation,
 }) => {
   const [isRenaming, setRenaming] = useState(false);
+  const [variationName, setVariationName] = useState<string>("");
+
+  const [error, setError] = useState<string | undefined>();
+
   const { setSlice } = useSliceState();
   const { saveSliceSuccess } = useSliceMachineActions();
 
+  useOnChange(isOpen, () => {
+    if (isOpen && variation?.name !== variationName) {
+      setVariationName(variation?.name ?? "");
```

Comment:
in other forms, we do this reset when the form is closed, not a biggie but maybe we want to use always the same approach?

Reply:
So this is the only form we do this, it's because the form is a bit different, and we don't always give the variation. So it means with the other way we would init with an empty value because it's not passed.,